File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& arr) {
        int mx = arr[0];
        int sum=arr[0];
        for(int i=1;i<arr.size();i++){
            sum=max(arr[i],sum+arr[i]);
            mx=max(mx,sum);
        }
        return mx;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] arr) {
        int r=0,sum=0,ans = Integer.MIN_VALUE;
        while(r<arr.length){
            sum+=arr[r++];
            ans =Math.max(ans,sum);
            if(sum<0)sum=0;
        }
        return ans;
    }
}